### PR TITLE
[4.3] Show message "No Media Found" for empty folder in Media Manager

### PR DESCRIPTION
**Pull request by rajputanuj31**
Pull Request for Issue #31340

### Summary of Changes
copied changes from previous PR #39518

### Testing Instructions
Create a new folder in Media and open it (No files should be present)

### Expected Result Before PR
No message

### Expected Result After PR
A Drag and drop icon and a message will show

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 1]
```diff
@@ -115,6 +118,9 @@ export default {
         width: this.$store.state.showInfoBar ? '75%' : '100%',
       };
     },
+    isEmpty() {
+      return ![...this.$store.getters.getSelectedDirectoryDirectories, ...this.$store.getters.getSelectedDirectoryFiles].length;
```
Comment (dgrammatiko) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]:
```suggestion
      return ![...this.$store.getters.getSelectedDirectoryDirectories, ...this.$store.getters.getSelectedDirectoryFiles].length
      && !this.$store.state.isLoading;
```

The compute `isEmpty` should be aware of the loading state and should not render if the data is not loaded

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by dgrammatiko** [comment]
@rajputanuj31 change https://github.com/joomla/joomla-cms/blob/cc4bdae6b7224d16c7aec31898fe9079e1407fd2/administrator/components/com_media/resources/scripts/components/browser/browser.vue#L20 to

```
v-if="listView === 'table' && !isEmpty"
```

change https://github.com/joomla/joomla-cms/blob/cc4bdae6b7224d16c7aec31898fe9079e1407fd2/administrator/components/com_media/resources/scripts/components/browser/browser.vue#L73 to

```
v-else-if="listView === 'grid'  && !isEmpty"
```

Also instead of 

```
<h4 v-if="isEmpty">
   {{ translate("COM_MEDIA_NO_MEDIA_FOUND") }}
 </h4>
```

we need something more to be inline with the other empty state pages, ie
Screenshot 2023-01-27 at 20 26 03

**Comment by laoneo** [comment]
A message is ok for now. We can do a followup with a more enhanced screen.

**Comment by rajputanuj31** [comment]
> A message is ok for now. We can do a followup with a more enhanced screen.

I have almost completed it, will make a commit soon.

**Comment by rajputanuj31** [comment]
Hey @dgrammatiko , @laoneo  Is this fine? also where can I find related documentation for this issue.

mediano

**Comment by laoneo** [comment]
Is this screen visible then for every folder which is empty? It can be that there is no documentation for it.

**Comment by rajputanuj31** [comment]
> Is this screen visible then for every folder which is empty? It can be that there is no documentation for it.

Yes it visible for every empty folders. Since no documentation for this issue so I will remove the **`Learn More`** link.

**Comment by dgrammatiko** [comment]
@rajputanuj31 sorry I miss guided you. What I meant was to refactor the drop state of the drag and drop to resemble the empty state for other components in Joomla. It doesn't have to be one to one the empty state of the other components
 
Screenshot 2023-01-28 at 12 03 41 (2)

**Comment by dgrammatiko** [comment]
@rajputanuj31 also please apply the requested changes above. Thanks

**Comment by rajputanuj31** [comment]
> @rajputanuj31 also please apply the requested changes above. Thanks

Here are the changes you've asked for.
image
Does this look good?

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 2]
```diff
@@ -70,7 +76,7 @@
         </tbody>
       </table>
       <div
-        v-else-if="listView === 'grid'"
+        v-else-if="listView === 'grid'  && !isEmpty"
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue]:
```suggestion
        v-else-if="listView === 'grid' && !isEmpty"
```

**Comment by rajputanuj31** [comment]
@dgrammatiko , @Quy , @laoneo please review my changes. Is there any more changes I need to make.

**Comment by dgrammatiko** [comment]
@rajputanuj31 personally I would go of something like

```
    <div
      v-if="isEmpty"
      class="text-center"
      style="display: grid; justify-content: center; align-content: center; height: 100%; margin-top: -1rem; color: var(--gray-200);"
    >
      <span
        class="fa-8x icon-cloud-upload upload-icon"
        aria-hidden="true"
      />
      <p>{{ translate("COM_MEDIA_DROP_FILE") }}</p>
    </div>
```
needs also 

```js
    /* The styles for the media-browser element */
    mediaBrowserStyles() {
      return {
        width: this.$store.state.showInfoBar ? '75%' : '100%',
        height: '100%',
      };
    },
```

Which results to a lighter icon/text and the icon and text are in the same position as the drag background.
Normal
Screenshot 2023-01-29 at 16 02 53

On drag
Screenshot 2023-01-29 at 16 03 02 2

**Comment by rajputanuj31** [comment]
@dgrammatiko after applying your changes icon remains at top but I made some changes using css those I am commiting
image

**Comment by dgrammatiko** [comment]
@rajputanuj31 did you replace also the `mediaBrowserStyles()` as I wrote above?

**Comment by rajputanuj31** [comment]
> @rajputanuj31 did you replace also the `mediaBrowserStyles()` as I wrote above?

yes

File: administrator/components/com_media/resources/scripts/components/browser/browser.vue [hunk 3]
```diff
@@ -115,6 +127,10 @@ export default {
         width: this.$store.state.showInfoBar ? '75%' : '100%',
```
Comment (dgrammatiko) [inline administrator/components/com_media/resources/scripts/components/browser/browser.vue:126]: Add `height: '100%',` after line 127, then use the code snippet I provided and the icon will be correctly centered

**Comment by laoneo** [comment]
Now we have a conflict here. Can you please check them, I guess the indentation of the tags has changed.

**Pull request closed without merging**